core/src/main/java/org/apache/calcite/sql2rel/StandardConvertletTable.java

```diff
@@ -2027,13 +2050,38 @@ private static class TimestampDiffConvertlet implements SqlRexConvertlet {
                 qualifier.getParserPosition());
         break;
       }
-      final RelDataType intervalType =
-          cx.getTypeFactory().createTypeWithNullability(
-              cx.getTypeFactory().createSqlIntervalType(qualifier),
-              op1.getType().isNullable() || op2.getType().isNullable());
-      final RexCall rexCall = (RexCall) rexBuilder.makeCall(
-          intervalType, SqlStdOperatorTable.MINUS_DATE,
-          ImmutableList.of(op2, op1));
+
+      RelDataType intervalType;
+      RexCall rexCall;
+      /* This additional logic handles the differing definitions of 'WEEK' between BigQuery
+      *  and Calcite. BigQuery considers Sunday as the start of the week, so two dates whose
+      *  day difference is <7 may have a week difference of 1 if they occur on two different
```

Comment:
Your comments seem to imply that the difference is due to weeks starting on a different day. I don't think that's true. Calcite considers Sunday to be the start of the week too. (Monday the start of an ISOWEEK.)

I think it's just that this particular function truncates before it subtracts.

Reply:
I will update the comment. What I was trying to say is, Calcite currently computes the number of weeks between two dates by dividing the number of days between them by 7 (and flooring the result). BigQuery however essentially checks how many Sundays are between the two dates, which in certain cases causes a discrepancy.